osh/hub/scripts/osh-worker-manager

```diff
@@ -0,0 +1,244 @@
+        delete_kobo_worker(kobo_worker)
+        sys.exit(move_worker_conf_status)
+    # Start the worker after this is finished
+    start_worker_cmd = f"ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=20 -i {settings.SINGLE_USE_WORKER_SSH_PRIVATE_KEY} {settings.SINGLE_USE_WORKER_SSH_USER}@{kobo_worker.name} 'sudo systemctl restart osh-worker'"
+    print(start_worker_cmd, file=sys.stderr)
+    start_worker_status = subprocess.call(start_worker_cmd, shell=True)
+    if start_worker_status != 0:
+        print(f"Failed to start worker worker at {kobo_worker.name}", file=sys.stderr)
+        delete_kobo_worker(kobo_worker)
+        sys.exit(start_worker_status)
+
+
+def create_worker(ip_address):
+    # Worker name should be set to hostname of the worker
+    worker_name = get_hostname_from_ip(ip_address)
+    try:
+        kobo_worker = create_worker_in_kobo(worker_name)
+        start_worker(kobo_worker)
+    except Exception as ex:  # noqa: B902
+        import traceback
+
+        # Ensure kobo worker is deleted in case of an exception.
+        # This would avoid spamming the database in a case continuous failures.
+        delete_kobo_worker(kobo_worker)
+
+        traceback.print_exception(ex, file=sys.stderr)
+        sys.exit(1)
+
+
+def delete_worker(ip_address):
+    from kobo.hub import models
+
+    # Worker name should be set to hostname of the worker
+    worker_name = get_hostname_from_ip(ip_address)
+    kobo_worker = models.Worker.objects.get(name=worker_name)
+    delete_kobo_worker(kobo_worker)
+
+
+def workers_needed():
+    import kobo.hub.models
+    from django.conf import settings
+
+    # Tasks in opened or assigned state, skip counting subtasks
+    # TODO: Check for only `mockbuild`, `diff-build` and `version-diff-build` tasks.
+    running_tasks = kobo.hub.models.Task.objects.running().filter(parent=None).count()
+
+    # Tasks waiting to be assigned, skip counting subtasks
+    # TODO: Check for only `mockbuild`, `diff-build` and `version-diff-build` tasks.
+    free_tasks = kobo.hub.models.Task.objects.free().filter(parent=None).count()
+
+    running_and_free_tasks = running_tasks + free_tasks
+
+    print(min(running_and_free_tasks, settings.MAX_SINGLE_USE_WORKERS))
+
+
+def check_finished(ip_address):
+    from kobo.hub import models
+
+    # Worker name should be set to hostname of the worker
+    worker_name = get_hostname_from_ip(ip_address)
+    kobo_worker = models.Worker.objects.get(name=worker_name)
+
+    # TODO: Check for stale workers that have max_load above 0, but do not have arch
+    # or channel set. They would never be used.
+
+    # TODO: Shall we perform further health checks here? For example, if `osh-worker`
+    # service is running on the worker machine?
+    if kobo_worker.max_load > 0:
+        print(f"{worker_name} is new.", file=sys.stderr)
+        sys.exit(1)
+
+    if kobo_worker.running_tasks().count() > 0:
```

Comment:
@lzaoral Would it be safer to use `task.is_finished()` (or some other) method here?

Reply:
I added a check for disabled workers, so this question is irrelevant now.